taker-frontend/src/useEventSource.ts

```diff
@@ -53,6 +58,8 @@ export function useEventSource(url: string, withCredentials?: boolean) {
     }, [source]);
 
     const toast = useToast();
+    // I dont think should be a hook.
+    // We should just conditionally render a component that displays this message
```

Comment:
If we want to keep using `useToast` we cannot use it inside a conditional.

Reply:
We can't call `useToast` conditionally but we can conditionally enable it by calling `toast()`!

Reply:
Isn't that what we are already doing?

Reply:
I think every call to `toast` creates a new one. If we want to update it we would need to `useRef`: https://chakra-ui.com/docs/feedback/toast#updating-toasts

There is also some talk about setting the `id` property of a toast to prevent duplicates.

Reply:
https://github.com/itchysats/itchysats/blob/master/maker-frontend/src/components/BackendMonitor.tsx#L31 - this is how we can reuse the same toast without creating duplicates

Reply:
I had something like what you guys are suggesting at some point, but I removed it because I never got duplicates when testing with the current version 🤷 I'm all for re-introducing that.